### ISSUE TITLE
Modifying Existing Page and Table Triggers using extension

### ISSUE THREAD

**Issue by gwaindepy** [MicrosoftDocs/dynamics365smb-devitpro-pb#1032]
Hi All, im a Pro in CAL but a newbie in AL. In CAL we used to modify existing code to control what happens when system triggers are called, for example I could change what happens in the OnInsertRecord page trigger. how do I do that in AL?

Example: there is a restriction in the OnInsertRecord Trigger on the item Journal Page. how do I bypass it

**Comment by MarcHansenMicrosoft** [comment]
Could you add more details on what restriction you have a problem with?

/Marc

**Comment by gwaindepy** [comment]
Hi Marc, here is a little background to my story

For the past 10 years i had a customisation from which i added 2 additional entry type to the item journal line table (issue and receipt) in addition to positive and negative adjustment.

So on the item journal page onInsert tigger there is code that restricts use of some entry types and its just out as code on the trigger.

I want to be able to allow my entry type to be processed through item journal page.

I am using D365 BC on premise so im running CAL and AL side by side because of the challenges i am facing. But i want to go AL full time.

In summary my challenges are

1) adding additional options to an option string in a table
2) setting table relationships for the options created
3) dealing with code that has no events to handle external code

**Comment by MarcHansenMicrosoft** [comment]
For:
1. You need to make a github request that you need the option to be changed to an enum.
2. I do not recall how much you can do the table relation from extension's. @atoader could you add something.
3. You need to make a github request for a new event.

**Comment by SusanneWindfeldPedersen** [comment]
Closing this issue.
